### PR TITLE
Feat/spasht 3.0

### PR DESCRIPTION
**Pull request by vivekaindia**
# Feat spasht for Definition UI and spasht db update
### PR depends on

> #1448 

## Description

Updated UI for definition and updated spasht db schema

### Changes

- Fixed schema fk name
- Integrated spasht db table
- updated UI for schedule agent
- added show details for definitions
- Fixed UI bug

## How to test

Describe the steps required to test the changes proposed in the pull request.

`sudo make install`

### CODE REVIEW

**Comment by vivekaindia** [comment]
@GMishx Please restart the check jobs for this PR. It contains a scheduler test error.

**Comment by GMishx** [comment]
@vivekaindia, please rebase with `feat/newagent/spasht` to resolve merge conflicts.

File: src/lib/php/Dao/SpashtDao.php [hunk 1]
```diff
@@ -1,21 +1,12 @@
...
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU General Public License for more details.
-
-You should have received a copy of the GNU General Public License along
-with this program; if not, write to the Free Software Foundation, Inc.,
-51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+Copyright 2019
+Author: Vivek Kumar<vvksindia@gmail.com>
+
+Copying and distribution of this file, with or without modification,
```
Comment (GMishx) [inline src/lib/php/Dao/SpashtDao.php]: Please use standard GPL-2.0-only license headers (earlier version was correct).

File: src/lib/php/Dao/SpashtDao.php [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    $params = [
+      $uploadID
+    ];
+
+    $sql = "SELECT * FROM spasht ".
+    "WHERE upload_fk = $1";
+
+    $row = $this->dbManager->getSingleRow($sql, $params, $statement);
+
+    /**
```
[the remark this answers is not in the document]
Comment (vivekaindia) [reply src/lib/php/Dao/SpashtDao.php]: fixed in feat/spasht_5.0

File: src/lib/php/Dao/SpashtDao.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
+<?php
+/*
+Copyright (C) 2008-2012 Hewlett-Packard Development Company, L.P.
+Copyright (C) 2014, Siemens AG
+Author: Johannes Najjar
```
[the remark this answers is not in the document]
Comment (vivekaindia) [reply src/lib/php/Dao/SpashtDao.php]: updated in feat/spasht_5.0

File: src/spasht/ui/Makefile [hunk 4]
```diff
@@ -1,20 +1,11 @@
...
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License along
-# with this program; if not, write to the Free Software Foundation, Inc.,
-# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
-######################################################################
+# Copying and distribution of this file, with or without modification,
```
[the remark this answers is not in the document]
Comment (vivekaindia) [reply src/spasht/ui/Makefile]: Updated in feat/spasht_5.0

File: src/spasht/ui/template/show_definitions.html.twig [hunk 5]
```diff
@@ -1,5 +1,44 @@
-<ul>
-    {% for key in body %}
-        <li>{{ key|e }}</li>
+{% if statusbody == "false" %}
```
[the remark this answers is not in the document]
Comment (vivekaindia) [reply src/spasht/ui/template/show_definitions.html.twig]: Added in feat/spasht_5.0

File: src/spasht/ui/template/show_uploads_table.html.twig [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<TABLE id="definitionsTable">
```
[the remark this answers is not in the document]
Comment (vivekaindia) [reply src/spasht/ui/template/show_uploads_table.html.twig]: Added in feat/spasht_5.0

File: src/lib/php/Dao/SpashtDao.php (hunk 1 above)
Comment (vivekaindia) [reply src/lib/php/Dao/SpashtDao.php]: updated in feat/spasht_5.0

File: src/spasht/ui/ui-spasht.php [hunk 7]
```diff
@@ -4,13 +4,18 @@
 use Fossology\Lib\Dao\UploadDao;
```
[the remark this answers is not in the document]
Comment (vivekaindia) [reply src/spasht/ui/ui-spasht.php]: updated in feat/spasht_5.0

File: src/spasht/ui/template/show_details_table.html.twig [hunk 8]
```diff
@@ -0,0 +1,46 @@
+<TABLE id="detailsTable" style="border: 1px solid black; border-collapse: collapse; display: none;">
```
[the remark this answers is not in the document]
Comment (vivekaindia) [reply src/spasht/ui/template/show_details_table.html.twig]: added in feat/spasht_5.0

File: src/spasht/agent/spasht.php [hunk 9]
```diff
@@ -1,21 +1,12 @@
 <?php
-/*
- Copyright (C) 2019
- Author: Vivek Kumar<vvksindia@gmail.com>
+# Copyright 2019
+# Author: Vivek Kumar<vvksindia@gmail.com>
+#
+# Copying and distribution of this file, with or without modification,
```
[the remark this answers is not in the document]
Comment (vivekaindia) [reply src/spasht/agent/spasht.php]: updated in feat/spasht_5.0

File: src/spasht/agent/SpashtAgent.php [hunk 10]
```diff
@@ -0,0 +1,39 @@
+<?php
+# Copyright 2019
+# Author: Vivek Kumar<vvksindia@gmail.com>
+#
+# Copying and distribution of this file, with or without modification,
```
[the remark this answers is not in the document]
Comment (vivekaindia) [reply src/spasht/agent/SpashtAgent.php]: updated in feat/spasht_5.0

File: src/spasht/ui/agent-spasht.php [hunk 11]
```diff
@@ -1,21 +1,12 @@
...
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License along
-# with this program; if not, write to the Free Software Foundation, Inc.,
-# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
-######################################################################
+# Copying and distribution of this file, with or without modification,
```
[the remark this answers is not in the document]
Comment (GMishx) [reply src/spasht/ui/agent-spasht.php]: resolved in Feat/spasht_5.0

File: src/spasht/ui/template/show_definitions.html.twig (hunk 5 above)
[the remark this answers is not in the document]
Comment (GMishx) [reply src/spasht/ui/template/show_definitions.html.twig]: I do not see them in Feat/spasht_5.0

File: src/spasht/ui/template/show_details_table.html.twig (hunk 8 above)
[the remark this answers is not in the document]
Comment (GMishx) [reply src/spasht/ui/template/show_details_table.html.twig]: I do not see them in Feat/spasht_5.0

File: src/spasht/ui/template/show_uploads_table.html.twig (hunk 6 above)
[the remark this answers is not in the document]
Comment (GMishx) [reply src/spasht/ui/template/show_uploads_table.html.twig]: I do not see them in Feat/spasht_5.0

File: src/spasht/ui/ui-spasht.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (GMishx) [reply src/spasht/ui/ui-spasht.php]: I do not see them in Feat/spasht_5.0

[the remark this answers is not in the document]
Comment (GMishx) [reply src/spasht/ui/ui-spasht.php]: The license header is still missing in Feat/spasht_5.0

Maybe we can do it later.

**Pull request merged**